app/models/miq_schedule.rb

```diff
@@ -428,4 +428,10 @@ def v_zone_name
     return "" if zone.nil?
     zone.name
   end
+
+  def self.find_by_report_id(report_id)
```

Comment:
It's more logical to have this two methods that are relative to report's schedules in the same place, but I don't really know where is the better place. IMHO this two methods will have more sens in report's class because those methods have more link with report's object. However the schedule's research can will be in schedule's class like a scope.